### PR TITLE
feat: Add support for AWS SES Template

### LINKED ISSUE #783: AWS::SES::Template

### PR DESCRIPTION
**Pull request by candiduslynx**
#### Summary

Implement AWS SES Template support (#783).
---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: resources/services/ses/templates.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			getInput := &ses.GetTemplateInput{TemplateName: templateMeta.Name}
+			template, err := svc.GetTemplate(ctx, getInput, func(o *ses.Options) { o.Region = c.Region })
+			if err != nil {
+				return diag.WrapError(err)
+			}
+			res <- &Template{
+				CreatedTimestamp: templateMeta.CreatedTimestamp,
+				Template:         template.Template,
+			}
+			<-time.After(time.Second) // wait for 1s
```
Comment (disq) [inline resources/services/ses/templates.go:101]: Why is this necessary? Same question with `nextList` below

File: client/data/partition_service_region.json [hunk 2]
```diff
@@ -4553,6 +4553,31 @@
             "us-west-2": {}
           }
         },
+        "ses": {
```
Comment (disq) [inline client/data/partition_service_region.json:4556]: Was this block added manually or from the region generator? (If manually, it will be overridden next time the generator runs)

File: docs/contributing/adding_a_new_resource.md [hunk 3]
```diff
@@ -26,11 +26,11 @@ If the service to which the resource belongs has not been used before in cq-prov
 ### Skeleton
 
 1. In [client/services.go](../../client/services.go), update the service interface and add the method(s) that you will be using to fetch the data from the aws sdk.
-1. Run `go generate client/services.go` to create a mock for your new methods. This will update [client/mocks/mock_<service>.go](../../client/mocks) automatically.
-1. Create a file under [resources/services/<service>](../../resources/services) that follows the pattern of `<resource>.go`.
+1. Run `go generate client/services.go` to create a mock for your new methods. This will update [client/mocks/mock_\<service\>.go](../../client/mocks) automatically.
```
Comment (disq) [inline docs/contributing/adding_a_new_resource.md:29]: Why was this doc changed, editor mishap? Is quoting necessary?

**Review by disq — COMMENTED** [review]
Some comments

File: resources/services/ses/templates.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	c := meta.(*client.Client)
+	svc := c.Services().SES
+
+	listInput := new(ses.ListTemplatesInput)
+	for {
+		output, err := svc.ListTemplates(ctx, listInput, func(o *ses.Options) { o.Region = c.Region })
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		nextList := time.After(time.Second)
```
Comment (irmatov) [inline resources/services/ses/templates.go:89]: I don't think we ever do such delays anywhere. In theory retries with backoffs should handle rate limiting but it's not ideal. Sleeping is not much better because it may slow down fetching significantly.

File: resources/services/ses/gen.hcl [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  column "template_html_part" {
+    rename = "html_part"
+  }
+  column "template_subject_part" {
+    rename = "subject_part"
+  }
+  column "template_text_part" {
+    rename = "text_part"
+  }
+  column "created_timestamp" {
```
Comment (irmatov) [inline resources/services/ses/gen.hcl:37]: nit: description is missing

**Review by irmatov — CHANGES_REQUESTED** [review]

File: client/data/partition_service_region.json (hunk 2 above)
Comment (candiduslynx) [reply client/data/partition_service_region.json:4556]: Manually. Could you point me to the generator?

File: resources/services/ses/templates.go (hunk 1 above)
Comment (disq) [reply resources/services/ses/templates.go:101]: If so, this isn't enough because of the multiplexer.

File: client/data/partition_service_region.json (hunk 2 above)
Comment (candiduslynx) [reply client/data/partition_service_region.json]: replaced with proper `email` entry

File: resources/services/ses/templates.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchSesTemplates(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().SES
+
+	listInput := new(ses.ListTemplatesInput)
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply resources/services/ses/templates.go:83]: `a := new(A)` ~ `a := &A{}`
I just find using `new` a bit cleaner

File: resources/services/ses/templates.go (hunk 4 above)
Comment (candiduslynx) [reply resources/services/ses/templates.go:89]: will remove the sleep (if the issue arises we can always put it back)

Comment (candiduslynx) [reply resources/services/ses/templates.go]: done

File: resources/services/ses/gen.hcl (hunk 5 above)
Comment (candiduslynx) [reply resources/services/ses/gen.hcl:37]: added

**Review by irmatov — APPROVED** [review]

**Review by disq — APPROVED** [review]
